TradeItIosTicketSDK2/TradeItYahooEquityTradingTicketViewController.swift

```diff
@@ -416,12 +416,29 @@ class TradeItYahooEquityTradingTicketViewController: TradeItYahooViewController,
         case .orderAction:
             cell.detailTextLabel?.text = orderCapabilities.labelFor(field: .actions, value: self.order.action.rawValue)
         case .quantity:
-            let cell = cell as? TradeItNumericToggleInputCell
+            let supportedOrderQuantityTypes = orderCapabilities.supportedOrderQuantityTypes(
+                forAction: self.order.action,
+                priceType: self.order.type
+            )
+            
+            if supportedOrderQuantityTypes.isEmpty {
+                let errorResult = TradeItErrorResult(
+                    title: "Order Quantity Type Error",
+                    message: "No valid quantity type found for this action",
+                    code: TradeItErrorCode.systemError
+                )
+                self.alertManager.showError(errorResult, onViewController: self)
+                return cell
+            }
+            
+            self.order.quantityType = supportedOrderQuantityTypes.contains(self.order.quantityType) ? self.order.quantityType : supportedOrderQuantityTypes[0]
```

Comment:
This code can still explode if someone accidentally removes the `isEmpty` check above. Better to use a guard?

```swift
guard let firstSupportedOrderQuantityType = supportedOrderQuantityTypes.first
    else {
        // show error
    }

self.order.quantityType = supportedOrderQuantityTypes.contains(self.order.quantityType) ? self.order.quantityType : firstSupportedOrderQuantityType
```

Reply:
Yeah the more swift-ey way of doing it